integration/network/delete_test.go

```diff
@@ -0,0 +1,80 @@
+package network
+
+import (
+	"context"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration/util/request"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func containsNetwork(nws []types.NetworkResource, nw types.NetworkCreateResponse) bool {
+	for _, n := range nws {
+		if n.ID == nw.ID {
+			return true
+		}
+	}
+	return false
+}
+
+// TestDockerNetworkDeletePreferID tests that if a network with a name
+// equal to another network's ID exists, the Network with the given
+// ID is removed, and not the network with the given name.
+func TestDockerNetworkDeletePreferID(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	n1, err := client.NetworkCreate(ctx, "testNW", types.NetworkCreate{})
+	require.NoError(t, err)
+	n2, err := client.NetworkCreate(ctx, n1.ID, types.NetworkCreate{})
+	require.NoError(t, err)
+
+	nws, err := client.NetworkList(ctx, types.NetworkListOptions{})
+	require.NoError(t, err)
+
+	assert.Equal(t, true, containsNetwork(nws, n1), "Network n1 not found")
+	assert.Equal(t, true, containsNetwork(nws, n2), "Network n2 not found")
+
+	// Delete the network using networkID. This should remove the original
+	// network, not the network with the name equal to the networkID
+	client.NetworkRemove(ctx, n1.ID)
+
+	nws, err = client.NetworkList(ctx, types.NetworkListOptions{})
+	require.NoError(t, err)
+	assert.Equal(t, false, containsNetwork(nws, n1), "Network n1 was not removed")
+	assert.Equal(t, true, containsNetwork(nws, n2), "Network n2 not found")
+}
+
+// TestDockerNetworkDeletePreferID tests that if a network with a name
+// equal to another network's partial ID exists, the Network with the given
+// Name is removed, and not the network with only a partially matching ID (prefix).
+func TestDockerNetworkDeletePreferFullName(t *testing.T) {
```

Comment:
This could be done in the same test case I think. If you create 3 networks and removes them in order of highest priority to lowest?